app/src/main/java/co/touchlab/kampkit/android/ui/Composables.kt

```diff
@@ -0,0 +1,202 @@
+import androidx.compose.animation.core.TweenSpec
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Divider
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.flowWithLifecycle
+import co.touchlab.kampkit.android.BreedViewModel
+import co.touchlab.kampkit.android.R
+import co.touchlab.kampkit.db.Breed
+import co.touchlab.kampkit.models.DataState
+import co.touchlab.kampkit.models.ItemDataSummary
+import co.touchlab.kermit.Kermit
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+
+@Composable
+fun MainScreen(
+    viewModel: BreedViewModel,
+    log: Kermit
+) {
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val lifecycleAwareDogsFlow = remember(viewModel.breedStateFlow, lifecycleOwner) {
+        viewModel.breedStateFlow.flowWithLifecycle(lifecycleOwner.lifecycle)
+    }
+    val dogsState by lifecycleAwareDogsFlow.collectAsState(viewModel.breedStateFlow.value)
+
+    MainScreenContent(
+        dogsState = dogsState,
+        onRefresh = { viewModel.refreshBreeds(true) },
+        onSuccess = { data -> log.v { "View updating with ${data.allItems.size} breeds" } },
+        onError = { exception -> log.e { "Displaying error: $exception" } },
+        onFavorite = { viewModel.updateBreedFavorite(it) }
+    )
+}
+
+@Composable
+fun MainScreenContent(
+    dogsState: DataState<ItemDataSummary>,
+    onRefresh: () -> Unit = {},
+    onSuccess: (ItemDataSummary) -> Unit = {},
+    onError: (String) -> Unit = {},
+    onFavorite: (Breed) -> Unit = {}
+) {
+    Surface(
+        color = MaterialTheme.colors.background,
+        modifier = Modifier.fillMaxSize()
+    ) {
+        val isRefreshingState by remember(dogsState) {
```

Comment:
I'm pretty sure this can be changed to just `isRefreshing = dogsState.loading` since its not doing any calculation, or just removed and do `rememberSwipeRefreshState(isRefreshing = dogsState.loading)` below